src/OpenTelemetry.Instrumentation.EntityFrameworkCore/Implementation/EntityFrameworkDiagnosticListener.cs

```diff
@@ -87,11 +90,19 @@ public override void OnCustom(string name, Activity activity, object payload)
 
                     var connection = this.connectionFetcher.Fetch(command);
                     var database = (string)this.databaseFetcher.Fetch(connection);
-
                     activity.DisplayName = database;
 
                     if (activity.IsAllDataRequested)
                     {
+                        try
```

Comment:
`Enrich` should be called at the end when the instrumentation library is done updating the Activity.